src/BaseFactory.php

```diff
@@ -265,9 +262,10 @@ protected function normalizeAttribute(mixed $value, string $name): mixed
     }
 
     /**
+     * @internal
      * @param Attributes $attributes
      */
-    final protected function mergedAttributes(array|callable $attributes): array
+    protected function mergedAttributes(array|callable $attributes): array
```

Comment:
this change is used to override `mergedAttributes` in `ObjectFactory`, in order to apply "before instantiate" callbacks